### PR TITLE
Adding Sub Account Endpoints

### PR DESCRIPTION
**Pull request by A-ndy-git**
Closes #  NA

## Summary

This PR adds all endpoints to interact with Binance Sub Accounts (excluding 4 referenced in `main-client.ts`). I completed this addition without the intention of publishing, but figured if it is in line with your continuous development, it may be of value.

If its not something you'd like to add at this time - let me know and I'll close the PR. :)

## Additional Information

This change adds a significant amount of new interfaces/types into `types/spot.ts`. I have tried to share and consolidate where possible, but it is still a large addition.

I performed some local testing, but have not written any test cases at this time.

##  Checks

- [ ] Updated related documentation
- [x] Updated/added related tests
- [x] I have tested my changes locally

### CODE REVIEW

**Review by MarcosNicolau — APPROVED** [review]

**Comment by MarcosNicolau** [comment]
Great Job! We really needed the typos for the Sub Accounts endpoints.

**Comment by tiagosiebler** [comment]
Spoke with Andy about this PR, some changes are still incoming. Mostly around type/method/variable naming.

**Comment by A-ndy-git** [comment]
Sorry for the delay gents. This PR now contains:

**Improved Naming Conventions**
**Final V2 Sub-Account Endpoints**

Please let me know if you have any concerns or suggestions. Cheers!

File: src/main-client.ts [hunk 1]
```diff
@@ -194,19 +242,150 @@ export class MainClient extends BaseRestClient {
    *
   **/
 
-  createVirtualSubAccount(params: VirtualSubaccountParams): Promise<any> {
+  createSubAccount(params: CreateSubAccountParams): Promise<VirtualSubAccount> {
```
Comment (tiagosiebler) [inline src/main-client.ts:245]:
```suggestion
  createVirtualSubAccount(params: CreateSubAccountParams): Promise<VirtualSubAccount> {
```

This is a breaking change (renaming an existing method that was here before the PR), can you change the method back to `createVirtualSubAccount`?

File: src/main-client.ts [hunk 2]
```diff
@@ -194,19 +242,150 @@ export class MainClient extends BaseRestClient {
...
     return this.postPrivate('sapi/v1/sub-account/virtualSubAccount', params);
   }
 
-  getSubAccountList(params?: SubAccountListParams): Promise<any> {
+  getSubAccountList(params?: GetSubAccountListParams): Promise<SubAccountListResponse> {
     return this.getPrivate('sapi/v1/sub-account/list', params);
   }
 
-  // TODO: https://binance-docs.github.io/apidocs/spot/en/#sub-account-endpoints
+  getSubAccountSpotAssetTransferHistory(params?: GetSubAccountSpotAssetTransferHistoryParams): Promise<SubAccountSpotAssetTransferHistory> {
```
Comment (tiagosiebler) [inline src/main-client.ts:253]:
```suggestion
  getSubAccountSpotAssetTransferHistory(params?: SubAccountSpotAssetTransferHistoryParams): Promise<SubAccountSpotAssetTransferHistory> {
```

Tiny nitpick (sorry), would prefer these interfaces to be without the "Get" prefix (following the standard set by the existing getter methods). Same for the other methods where the param interface starts with `Get`

**Review by tiagosiebler — COMMENTED** [review]
Sorry for the delay - looks great to me, thanks for working on this! Two tiny nitpicks and then I think it's ready. Again, sorry it took so long to look over this.

**Review by tiagosiebler — APPROVED** [review]
LGTM, thanks!

**Pull request merged**